### PR TITLE
Add AssetBundle::cssStrings + extract position from JS/CSS options + Refactoring

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #70 

- Add `AssetBundle::cssStrings`
- Extract position from options to `AssetBundle::cssPosition` and `AssetBundle::jsPosition`
- Add `AssetLoader::withCssDefaultPosition` and `AssetLoader::withJsDefaultPosition`
- Options used both for JS/CSS files and for JS/CSS blocks
- Position used for JS/CSS files, JS/CSS blocks and JS variables
- Improve phpdoc

Example of CSS files (for JS similar):

```php
public array $css = [
	'https://cdn.jsdelivr.net/npm/bulma@0.9.2/css/bulma.min.css',
	'css/main.css',
	['css/a.css'],
	['css/b.css', 3],
	['css/c.css', 3, 'crossorigin' => 'any'],
	'key' => 'css/d.css',
]
```

Example of JS strings (for CSS similar):

```php
public array $jsStrings = [
	'alert(1);',
	['alert(2);'],
	['alert(3);', 3],
	['alert(4);', 3, 'id' => 'main'],
	'key1' => 'alert(5);',
	'key2' => ['alert(6);'],
	'key3' => ['alert(7);', 3],
	'key4' => ['alert(8);', 3, 'id' => 'second'],
];
```

Example of JS variables:

```php
public array $jsVars = [
	'var1' => 'value1',
	['var2', 'value2', 3],
];
```

### CODE REVIEW

File: docs/asset-bundles.md [hunk 1]
```diff
@@ -22,19 +22,25 @@ Combinations of these types could be used as well.
...
-`$sourcePath`    |`string/null`| `null`  | The directory that contains the source asset files for this asset bundle.
+  Name              | Type       | Default | Description
+------------------- |-------------|---------|------------
+`$basePath`         |`string|null`| `null`  | The web public directory that contains the asset files in this bundle.
+`$baseUrl`          |`string|null`| `null`  | The base URL for the relative asset files listed in `$js` and `$css`.
+`$cdn`              |`bool`       | `false` | Indicates if we are going to use CDN exclusively.
+`$css`              |`array`      | `[]`    | List of CSS files that this bundle contains.
+`$cssOptions`       |`array`      | `[]`    | The options that will be passed to `\Yiisoft\View\WebView::setCssFiles()`.
+`$cssStrings`       |`array`      | `[]`    | List of CSS blocks.
+`$cssPosition`      |`int|null`   | `null`  | CSS files and blocks position.
```
Comment (devanych) [inline docs/asset-bundles.md:33]: Probably need to specify in docblock what `int ' stands for, what possible values are.

File: docs/asset-bundles.md [hunk 2]
```diff
@@ -22,19 +22,25 @@ Combinations of these types could be used as well.
...
+`$converterOptions` |`array`      | `[]`    | The options line command from converter
+`$depends`          |`array`      | `[]`    | List of bundle class names that this bundle depends on.
+`$js`               |`array`      | `[]`    | List of JavaScript files that this bundle contains.
+`$jsStrings`        |`array`      | `[]`    | List of JavaScript blocks.
+`$jsOptions`        |`array`      | `[]`    | The options that will be passed to `\Yiisoft\View\WebView::setJsFiles()`.
+`$jsPosition`       |`int|null`   | `null`  | JavaScript files and blocks position.
```
Comment (devanych) [inline docs/asset-bundles.md:39]: Probably need to specify in docblock what `int ' stands for, what possible values are.

File: src/AssetBundle.php [hunk 3]
```diff
@@ -42,19 +47,79 @@ class AssetBundle
...
+    public array $cssStrings = [];
+
     /**
      * @var array The options that will be passed to {@see \Yiisoft\View\WebView::registerCssFile()}
      * when registering the CSS files in this bundle.
      */
     public array $cssOptions = [];
 
+    /**
+     * @var int|null CSS files and blocks position.
```
Comment (devanych) [inline src/AssetBundle.php:119]: Probably need to specify in docblock what `int ' stands for, what possible values are.

File: src/AssetBundle.php [hunk 4]
```diff
@@ -105,22 +170,66 @@ class AssetBundle
...
+
     /**
      * @var array The options that will be passed to {@see \Yiisoft\View\WebView::registerJsFile()}
      * when registering the JS files in this bundle.
      */
     public array $jsOptions = [];
 
     /**
-     * @var array JavaScript code blocks to be passed to {@see \Yiisoft\View\WebView::registerJs()}.
+     * @var int|null JavaScript files and blocks position.
```
Comment (devanych) [inline src/AssetBundle.php:214]: Probably need to specify in docblock what `int ' stands for, what possible values are.

File: src/AssetLoader.php [hunk 5]
```diff
@@ -226,6 +233,13 @@ public function withCssDefaultOptions(array $cssDefaultOptions): self
         return $new;
     }
 
+    public function withCssDefaultPosition(?int $position): self
```
Comment (devanych) [inline src/AssetLoader.php:236]: Probably need to specify in docblock what `int ' stands for, what possible values are.

File: src/AssetLoader.php [hunk 6]
```diff
@@ -241,6 +255,13 @@ public function withJsDefaultOptions(array $jsDefaultOptions): self
         return $new;
     }
 
+    public function withJsDefaultPosition(?int $position): self
```
Comment (devanych) [inline src/AssetLoader.php:258]: Probably need to specify in docblock what `int ' stands for, what possible values are.

File: docs/asset-bundles.md (hunk 1 above)
Comment (vjik) [reply docs/asset-bundles.md]: Done.

File: docs/asset-bundles.md (hunk 2 above)
Comment (vjik) [reply docs/asset-bundles.md]: Done.

File: src/AssetBundle.php (hunk 3 above)
Comment (vjik) [reply src/AssetBundle.php]: Done.

File: src/AssetBundle.php (hunk 4 above)
Comment (vjik) [reply src/AssetBundle.php]: Done.

File: src/AssetLoader.php (hunk 5 above)
Comment (vjik) [reply src/AssetLoader.php:241]: Done.

File: src/AssetLoader.php (hunk 6 above)
Comment (vjik) [reply src/AssetLoader.php:268]: Done.

**Review by devanych — APPROVED** [review]
Suggestions for the future PR:

1. The `AssetManager` class has become complex and will be difficult to maintain. Need to think about how to divide it into responsibilities. For example, adding an internal `AssetRegistarator` and put all the CSS and JS manipulations in it.

2. I don't like that we have an implicit dependency on `yiisoft/view`. Namely, by positions (`WebView:: POSITION_HEAD`, `WebView::POSITION_END` , etc.). Need to think about how to get away from this.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**